### PR TITLE
min fix of 'storage_type' index mismatch

### PR DESCRIPTION
**Pull request by gssplayer**
Index of array 'storage_type' should be indice of 'node entries',
not 'nodes'.

sign-of-by: songshan.gong@horizon.ai

## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by abhinavs95** [comment]
Thank you for the fix @gssplayer Could you have a look at the CI failure?

@mxnet-label-bot add [Backend, pr-awaiting-review]

**Comment by piyushghai** [comment]
@gssplayer Can you look into the CI failures in order to take this PR forward ?

**Comment by reminisce** [comment]
Please rebase and retrigger CI.

File: src/executor/graph_executor.cc [hunk 1]
```diff
@@ -947,8 +947,8 @@ void GraphExecutor::InitDataEntryMemory(std::vector<NDArray>* shared_pool) {
     for (uint32_t i = 0; i < idx[nid].source->num_outputs(); ++i) {
       auto eid = idx.entry_id(nid, i);
       data_context[eid] = vctx[nid];
-      CHECK_NE(vstorage_type[nid], kUndefinedStorage);
-      data_storage_type[eid] = (NDArrayStorageType) vstorage_type[nid];
```
Comment (eric-haibin-lin) [inline src/executor/graph_executor.cc]: Thanks for the fix! Is there any failed unit test you have at hand?

**Comment by roywei** [comment]
@gssplayer Thanks for the contribution, could you rebase your PR and trigger CI test again?

File: src/executor/graph_executor.cc (hunk 1 above)
Comment (gssplayer) [reply src/executor/graph_executor.cc]: I'am very sorry for no response on this question. I found this problem just by reading code, and had no any test. I've no runtime enviroment at hand. And I have to admit your CI is too complicate for me. Can anyone help to tracking this problem?

**Comment by abhinavs95** [comment]
@gssplayer Could you have a look at the CI failure? You could try re-triggering it.

@mxnet-label-bot update [Backend, pr-awaiting-response]

**Comment by abhinavs95** [comment]
@mxnet-label-bot update [Backend, pr-awaiting-response]

**Comment by eric-haibin-lin** [comment]
Triggered. hopefully it passes

**Comment by piyushghai** [comment]
@gssplayer The CI passed :)

@eric-haibin-lin Can this PR be merged if it looks good ?

**Pull request merged**